src/core/server.ts

```
@@ -79,7 +99,19 @@ export class Server {
   }
 
   getLifeCycleStatus(): Readonly<ServerLifeCycle[]> {
-    return this.lifeCycle;
+    return this.lifeCycleHistory;
+  }
+
+  private removePendingSchema() {
+    const fetchingPos = this.lifeCyclePending.indexOf(ServerLifeCycle.REMOTE_SCHEMAS_FETCHING);
```

If this doesn't find the lifecycle constant, the splice method will fail. But alternatively, you could explicitly check for it returning -1, and decide how to proceed. Likely you'd still want to throw an error since this item not being in the list seems like it would be an illegal programming error rather than an unexpected but still possible result of an external event, but it would give you the flexibility to handle the problem more gracefully. At the very least, if you do still throw a hard error, you'd be able to fail closer to the actual problem site and tailor the message so that the problem is much easier to debug and fix.